week 1/lists_n_loops/driver.py

```diff
@@ -0,0 +1,77 @@
+import index_search as test
+import time
+import matplotlib.pyplot as plt
+import random as rnd
+
+if __name__ == "__main__":
+
+    while True:
+
+        # input number of elements
+        size_n = int(input("Enter the number of elements in the list: "))
+
+        nums = [rnd.randrange(1, 100) for i in range(size_n)]
+
+        # Asking for the key to be searched
+        key = int(input("Enter the number "
+                        "you want to search : "))
+
+        # Calculating Execution of first loop
+        start = time.time()
+        test.search_key_first(nums, key=key)
+        end = time.time()
+        first_exec_time = end - start
+        print(f"Time taken to for the "
```

Comment:
you can create a helper function or decorator that returns the execution time for a method or particular lines of code. Here, you are re-writing the same sort of code again and again for all 5 variants.